File: 백준/Silver/11650. 좌표 정렬하기/좌표 정렬하기.java
import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;
import java.util.*;

public class Main {
    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        StringBuilder sb = new StringBuilder();
        int n = Integer.parseInt(br.readLine());

        StringTokenizer st;
        List<Point> list = new ArrayList<>();
        for (int i = 0; i < n; i++) {
            st = new StringTokenizer(br.readLine());

            int x = Integer.parseInt(st.nextToken());
            int y = Integer.parseInt(st.nextToken());

            list.add(new Point(x, y));
        }

        Collections.sort(list);
        for (Point p : list) {
            sb.append(p.x).append(' ').append(p.y).append('\n');
        }

        System.out.println(sb.toString());
    }

    static class Point implements Comparable<Point> {
        int x;
        int y;

        public Point(int x, int y) {
            this.x = x;
            this.y = y;
        }

        @Override
        public int compareTo(Point o) {
            int result = this.x - o.x;
            if (result == 0) {
                result = this.y - o.y;
            }

            return result;
        }
    }
}

File: 백준/Silver/11650. 좌표 정렬하기/좌표 정렬하기.kt
import java.io.*
import java.util.*

data class Point(
    val x: Int,
    val y: Int
) : Comparable<Point> {
    override fun compareTo(other: Point): Int {
        val result = x - other.x;
        return if (result == 0) y - other.y else result
    }
}

fun main() = with(BufferedReader(InputStreamReader(System.`in`))) {
    val n = readLine().toInt()
    val points = Array(n) {
        val st = StringTokenizer(readLine())

        val x = st.nextToken().toInt()
        val y = st.nextToken().toInt()
        Point(x, y)
    }

    points.sort()

    val sb = StringBuilder()
    for (p in points) {
        sb.append(p.x).append(' ').append(p.y).append('\n')
    }

    print(sb);
}